test/metrics/breakdown.js

```diff
@@ -0,0 +1,596 @@
+    agent._instrumentation.activeSpan = span0
+
+    var span1 = agent.startSpan('SELECT *', 'db.mysql', { startTime: 15, childOf: span0 })
+    if (span0) span0.end(20)
+    if (span1) span1.end(25)
+    transaction.end(null, 30)
+  })
+
+  t.test('with app sub-span extending beyond end', t => {
+    resetAgent(5, conf, ({ metricsets }) => {
+      const found = {
+        transaction: finders.transaction(metricsets),
+        transaction_span: finders['transaction span'](metricsets)
+      }
+
+      t.ok(found.transaction, 'found transaction metricset')
+      t.deepEqual(found.transaction.samples, {
+        'transaction.duration.count': { value: 1 },
+        'transaction.duration.sum.us': { value: 20 },
+        'transaction.breakdown.count': { value: 1 }
+      }, 'sample values match')
+
+      t.ok(found.transaction_span, 'found app span metricset')
+      t.deepEqual(found.transaction_span.samples, {
+        'span.self_time.count': { value: 1 },
+        'span.self_time.sum.us': { value: 10 }
+      }, 'sample values match')
+
+      t.notOk(finders.span(metricsets, { type: 'db.mysql' }), 'does not have un-ended spans')
+
+      agent._metrics.stop()
+      t.end()
+    })
+
+    var transaction = agent.startTransaction('foo', 'bar', { startTime: 0 })
+    var span0 = agent.startSpan('foo', 'app', { startTime: 10 })
+
+    // Hack to make it look like an async tick has already happened
+    agent._instrumentation.activeSpan = span0
+
+    transaction.end(null, 20)
+    var span1 = agent.startSpan('SELECT *', 'db.mysql', { startTime: 20, childOf: span0 })
+    if (span0) span0.end(30)
+    if (span1) span1.end(30)
+  })
+
+  t.test('with other sub-span extending beyond end', t => {
+    resetAgent(4, conf, ({ metricsets }) => {
+      const found = {
+        transaction: finders.transaction(metricsets),
+        transaction_span: finders['transaction span'](metricsets)
+      }
+
+      t.ok(found.transaction, 'found transaction metricset')
+      t.deepEqual(found.transaction.samples, {
+        'transaction.duration.count': { value: 1 },
+        'transaction.duration.sum.us': { value: 20 },
+        'transaction.breakdown.count': { value: 1 }
+      }, 'sample values match')
+
+      t.ok(found.transaction_span, 'found app span metricset')
+      t.deepEqual(found.transaction_span.samples, {
+        'span.self_time.count': { value: 1 },
+        'span.self_time.sum.us': { value: 10 }
+      }, 'sample values match')
+
+      t.notOk(finders.span(metricsets, { type: 'db.mysql' }), 'does not have un-ended spans')
+
+      agent._metrics.stop()
+      t.end()
+    })
+
+    var transaction = agent.startTransaction('foo', 'bar', { startTime: 0 })
+    var span = agent.startSpan('SELECT *', 'db.mysql', { startTime: 10 })
+    transaction.end(null, 20)
+    if (span) span.end(30)
+  })
+
+  t.test('with other sub-span starting after end', t => {
+    resetAgent(5, conf, ({ metricsets }) => {
```

Comment:
should be 4 I think: 3 metricsets + transaction
(no span because it starts after transaction)

Reply:
fixed